src/main/kotlin/com/github/lumber1000/simex/clients/RandomTrader.kt

```diff
@@ -1,60 +1,79 @@
+import java.util.concurrent.*
 
 class RandomTrader(
     host: String,
     port: Int,
-    logger: Logger,
-    private val barrier: CyclicBarrier,
     private val ticker: String,
     private val orderType: OrderType,
-    private val random: Random
-    ) : BaseClient(host, port, logger), Runnable {
+    private val random: Random,
+    private var numberOfOrders: Int
+) : BaseClient(host, port) {
 
-    override fun run() {
-        waitForConnection()
+    private suspend fun placeOrders() {
+        // waitForConnection()
 
-        repeat(20) {
+        repeat(numberOfOrders) {
             val order = Order(0L, orderType, ticker, random.nextInt(100, 200), random.nextInt(1, 100), 0L)
             stubRxAdapter.submitOrder(order)
-            logger.info("Send order: $order")
-            Thread.sleep(random.nextLong(200, 800))
+            LOGGER.info("Send order: $order")
+            delay(random.nextLong(200, 800))
         }
+    }
 
-        barrier.await()
+    private suspend fun cancelUnsatisfiedOrders() {
+        var book = stubRxAdapter.getBook(listOf(ticker)).blockingGet()
+        if (book.isEmpty()) return
 
-        if (orderType == OrderType.BUY_LIMIT) {
-            var book = stubRxAdapter.getBook(listOf(ticker)).blockingGet()
-            println("\nOrderBook for $ticker (${book.size} orders)\n\n$book\n")
-            println("\n\nCancel all orders\n")
+        println("\nOrderBook for $ticker (${book.size} orders):\n\n$book\n")
+        println("Cancel unsatisfied orders:\n")
 
-            book.forEach {
-                logger.info("Cancel order: $it")
-                stubRxAdapter.cancelOrder(it)
-            }
+        book.forEach {
+            LOGGER.info("Cancel order: $it")
+            stubRxAdapter.cancelOrder(it)
+        }
+
+        delay(10)
+
+        book = stubRxAdapter.getBook(listOf(ticker)).blockingGet()
+        println("\nOrderBook for $ticker (${book.size} orders)\n\n$book\n")
+    }
+
+    companion object {
+        private val LOGGER = KotlinLogging.logger {}
+
+        @JvmStatic
+        fun main(vararg args: String) {
+            setDefaultExceptionHandler(LOGGER)
+            val config = loadConfig()
+            val numberOfOrders = 20
+
+            fun createTrader(ticker: String, type: OrderType, seed: Int) =
+                RandomTrader(config.hostname, config.port, ticker, type, Random(seed), numberOfOrders)
 
-            book = stubRxAdapter.getBook(listOf(ticker)).blockingGet()
-            println("\n\nOrderBook for $ticker (${book.size} orders)\n$book\n")
+            val traders = listOf(
+                createTrader("T1", OrderType.BUY_LIMIT, 1000),
+                createTrader("T1", OrderType.SELL_LIMIT, 2000),
+                createTrader("T2", OrderType.BUY_LIMIT, 3000),
+                createTrader("T2", OrderType.SELL_LIMIT, 4000)
+            )
+
+            Executors.newSingleThreadExecutor().asCoroutineDispatcher().use { context ->
```

Comment:
Redundant construction, because the runBlocking method already runs a new coroutine